wumpus_world.py

```diff
@@ -118,21 +117,154 @@ def _transform_kb(self):
                     facts.add(sentence)
         
         return facts, implications, bidirectionals
+    
+    def inference_by_resolution(self, query):
+        """
+        This will be the main method that you will call to
+        perform inference by resolution.
+        """
+        # CNF The kb and add the negated query to it
+        kb_with_negated_query = [self.sentence_to_cnf(i) for i in self.kb]
+        kb_with_negated_query.append(('NOT', self.sentence_to_cnf(query)))
+        
+        while True:
+            # Comparing each clause with every other clause in the kb
+            for i in range(len(kb_with_negated_query) - 1):
+                for j in range(i + 1, len(kb_with_negated_query)):
+                    resolved = self.resolve(kb_with_negated_query[i], kb_with_negated_query[j])
+
+                    # Check if this is the last possible pair of clauses, if it is we return True when the kb contradicts itself and False when it doesn't
+                    if set(resolved).issubset(set(kb_with_negated_query)) and i + 2 == len(kb_with_negated_query) and j + 1 == len(kb_with_negated_query):
+                        if query in kb_with_negated_query and ('NOT', query) in kb_with_negated_query:
+                            return True
```

Comment:
My intuition is that this works too, but normally, we just want to check if all clauses resolve to an empty set. If you think about it, what you are checking here - whether the clause set contains two complementary literals - can be part of resolution itself, right? That would give you an empty set, which you can check here.